modules/apps/address/address-impl/src/main/java/com/liferay/address/internal/search/spi/model/query/contributor/AddressModelPreFilterContributor.java

```diff
@@ -80,7 +88,36 @@ private void _filterByTypeId(
 			return;
 		}
 
-		long[] typeIds = (long[])params.get("typeIds");
+		long[] typeIds = (long[])params.getOrDefault("typeIds", new long[0]);
+		String[] typeNames = (String[])params.get("typeNames");
+
+		if (ArrayUtil.isNotEmpty(typeNames)) {
+			long classNameId = GetterUtil.getLong(
+				searchContext.getAttribute(Field.CLASS_NAME_ID));
+
+			if (classNameId > 0) {
+				ClassName className = _classNameLocalService.fetchByClassNameId(
+					classNameId);
+
+				if (className != null) {
+					List<Long> listTypeIds = new ArrayList<>();
+
+					for (String typeName : typeNames) {
+						ListType listType = _listTypeLocalService.getListType(
+							typeName,
+							className.getClassName() +
+								ListTypeConstants.ADDRESS);
```

Comment:
Ok so I don't have a firm opinion on how we should or should not handle the error here if a `ListType` is not found. We could:

1. Allow it to throw and stop the search
2. Catch inside the loop and log an error, still allowing the search to continue.

Thoughts, @pei-jung ?

Reply:
Also, this may be a good spot to use `stream` in order to reduce the nested `if` statements

Reply:
Good catch. `listTypeLocalService.getListType` doesn't throw an exception but we should check for null value. We can skip it if a ListType is not found since it's an invalid value.

I'll add the null check and update it to use `stream`.